doc/git_submodule.md

```diff
@@ -0,0 +1,70 @@
+# Using git submodules to develop a game
+
+Caveat: This is not necessarily the one true way. The framework and its practices
+are constantly being developed, but this document contains some pointers.
+
+If someone wants to add specific git commands here, please do. The current author
+will not bother, because formatting and anonymizing the repository URLs is too
+much work. This is the downside of open source, you do not always get everything
+for free.
+
+## Why?
+
+Git's strength is in how well it handles remotes and branches. This may go beyond
+Git 101, but it's a very efficient way to work once you get into it.
+
+Having the framework separate from your game also helps to identify changes to the
+code you may have to make. The goal of a framework is to be completely separate from
+whatever it's used for, so please open issues on GitHub if you find yourself making
+changes to the framework that aren't bugfixes or new features.
+
+If possible, please make a PR that alters Escoria's behavior, eg. through `ProjectSettings`,
+so that the change doesn't happen in Escoria's upstream code.
+
+In a wider sense, please expose a new internal API to extend the functionality
+of the framework instead of hacking your own fork. This will make your code more maintanable
+and make Escoria all the more better. An example of this could be a mechanism for implementing
+custom ESC script commands. This does not exist at the time of writing this.
+
+## How?
+
+Create a git repository and add the Escoria GitHub URL as a remote, eg. `gh-escoria`.
+This document will refer to your repository as `origin`.
+
+Reset your `master` to match whatever `gh-escoria/master` is at the time.
+
+Generally messing with `master` is discouraged. You may adapt whatever workflow you want
+to this, but this document will use `devel`.
+
+The project root is `device/`. It contains your project settings. When submitting branches
+for merging, please do not include your game's settings. Any new features should be disabled
+by default, to not mess with the tutorial [*Creating Point and Click Games with Escoria*](https://fr.flossmanuals.net/creating-point-and-click-games-with-escoria/).
+
+## Actual work
+
+Create a new repository for the submodule.
+
+Add it as `device/game/`.
```

Comment:
Add your new directory to a `.gitignore` file at the root of your directory and initialize a new git repository under `device/game` with the command `git init`.

Reply:
Is this how you would write it? I don't really know why it'd be in `.gitignore` (whose most useful contents seem to be `*.import`), but I never tried it.

It's valuable for the branches to be able to track the submodule. Like in the trying-to-sort-out-inventory work. I have a devel branch in Escoria for the work and an "inventory" branch in my submodule, and the devel branch has committed the submodule to the correct ("inventory") branch and everything works as expected.

Maybe because I never tried ignoring a submodule, I'm missing out on something, so would you please elaborate?

Reply:
The reason for the confusion is that I didn't know sub modules was an actual feature of Git until just now, so I've always just used ad-hoc sub modules, where I work with the two repositories independently. I think this is likely to be too difficult for your average game developer to follow, so I'm leaning towards just providing advice on separating game from framework, since that would be useful, even if they are just using zip-archives for "version control", as some do.

Reply:
Ah! That does explain it! Ok, we're addressing different audiences. I'm addressing the audience I represent myself, the one who's familiar and somewhat comfortable with this submodule-style stuff, but doesn't know how to apply it to Escoria.

IMO the non-submodule approach is by its very nature - literally it could be anything but a submodule - something that would be outside the scope of this document. The average game developer can read that other document instead.